### ISSUE TITLE
Add Quality Assurance features

### ISSUE THREAD

**Issue by mathieuboudreau** [axondeepseg/axondeepseg#760]
This is a master issue to discuss quality assurance features, workflows, and ideas. Below we'll list and update issues or PRs that relate to QA, please edit as needed.

## Issues
- [ ] #779
- [ ] Napari - display histograms images produced above in viewer after segmentation
- [ ] Find a workflow that allows for quick removal of entire axon/myelin "objects" after a napari segmentation run, maybe by clicking/submitting a list of IDs/automatic removal for suspected outliers along with a map of them in a separate image.

**Comment by mathieuboudreau** [comment]
Add a "% probability" or severity that an axon is taken at an oblique cut (myelin thickness on minor axis divided by myelin thickness on major axis)

**Comment by hermancollin** [comment]
Add the option to save uncertainty maps from the raw output of the model. This was originally suggested by Brad Zuchero. This would give an idea of how confident the model is about some regions. Maybe some smart thresholding could enhance segmentations in some cases. Related to https://github.com/axondeepseg/axondeepseg/issues/340. Also, it would be very easy to implement (see https://github.com/axondeepseg/axondeepseg/discussions/549, which might be a bit outdated).

**Comment by hermancollin** [comment]
> Add a "% probability" or severity that an axon is taken at an oblique cut (myelin thickness on minor axis divided by myelin thickness on major axis)

Related to this: maybe we could re-use an idea used in AxonSeg. Once the axons were segmented and morphometrics were extracted, they used a classifier to reject shapes that were likely not axons. This classifier was a simple ML model that rejected axons based on these features:
- Circularity
- Solidity
- Eccentricity
- Equivalent diameter
- Object intensity
- Neighborhood intensity
- Contrast

Maybe a similar project could help clean up masks based on this idea. The annotations would be somewhat easy to collect (e.g. segment a bunch of imgs in different contrasts and manually choose bad axon IDs). Also this could easily be a "self-contained" internship project.

**Comment by hermancollin** [comment]
An idea from Stanford and McGill collaborators: plot metrics in "bin" (e.g. grouping by axon diameter ranges). For example, when comparing the mean g-ratio from 2 distributions, the mean value could be identical, but differences could be significant between big/small axons.

**Comment by mathieuboudreau** [comment]
Great idea! Noted

**Comment by hermancollin** [comment]
@mathieuboudreau we need to generate histograms by "bins" of axon diameters for the project with mcgill/ulaval. Instead of doing it only for them, I want to implement it in ADS so we can do it automatically and conveniently in the future.

Do you think I should work on a branch of #781, directly in #781 or independently of it? @MurielleMardenli200 and I planned on working on it this week.

**Comment by mathieuboudreau** [comment]
>Do you think I should work on a branch of https://github.com/axondeepseg/axondeepseg/pull/781, directly in https://github.com/axondeepseg/axondeepseg/pull/781 or independently of it? @MurielleMardenli200 and I planned on working on it this week.

I'd say maybe branch off of #781, and then make a PR to merge it into there? Any way you suggested works though, whatever is convenient for you! Thanks!

**Comment by hermancollin** [comment]
Yes that's right I guess it's less of a QA feature. We'll work on this independently then.
